crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -457,6 +482,7 @@ impl<'w> UnsafeWorldCell<'w> {
     /// - no other references to the resource exist at the same time
     #[inline]
     pub unsafe fn get_resource_mut<R: Resource>(self) -> Option<Mut<'w, R>> {
+        self.assert_allows_mutable_access();
```

Comment:
You don't technically need this one (or the one in `get_non_send_resource_mut`), since it calls `get_resource_mut_by_id` which already does the check.  

Or, if you want to do it consistently this way, you're missing a few like `UnsafeEntityCell::get_mut`.

Reply:
Ah good point, this call is unnecessary